src/components/application_manager/src/application_data_impl.cc

```diff
@@ -568,30 +568,37 @@ void DynamicApplicationDataImpl::set_display_capabilities(
   display_capabilities_.reset(
       new smart_objects::SmartObject(display_capabilities));
 
-  auto has_window_id = [&tmp_window_capabilities](const WindowID window_id) {
+  auto has_window_id = [&tmp_window_capabilities](const WindowID window_id,
+                                                  int& found_index) {
     const auto tmp_window_capabilities_arr = tmp_window_capabilities.asArray();
     if (!tmp_window_capabilities_arr) {
+      found_index = -1;
       return false;
     }
 
     for (auto element : *tmp_window_capabilities_arr) {
       if (element.keyExists(strings::window_id)) {
         if (window_id == element[strings::window_id].asInt())
           return true;
-      } else if (window_id == 0) {
+      } else if (window_id == 0 || window_id == -1) {
```

Comment:
I assume that `-1` is what is returned by `incoming_window_capabilities[i][strings::window_id].asInt()` if the key isn't present? I think it would be better to check `keyExists` rather than blindly using `asInt()` if possible.
Maybe:
```
auto window_id = 0;
if (incoming_window_capabilities[i].hasKey(strings::window_id)) {
  window_id = incoming_window_capabilities[i][strings::window_id].asInt();
}
...
```